src/web/src/hooks/use-telescope-contributor.ts

```diff
@@ -12,31 +11,80 @@ export type Contributor = {
 const telescopeGitHubContributorsUrl =
   'https://api.github.com/repos/Seneca-CDOT/telescope/contributors';
 
-const randomContributor = (contributors?: any[]) =>
-  contributors ? contributors[Math.floor(Math.random() * contributors.length)] : null;
+const prepareURL = (page?: number) =>
+  `${telescopeGitHubContributorsUrl}?per_page=1&${page ? `page=${page}` : ''}`;
 
-const PER_PAGE: number = 50;
+const linkParser = (links: string, key: string) => {
+  let url;
+  let page;
+  const regex = new RegExp(`<([^?]+\\?per_page=1&[a-z]+=([\\d]+))>;[\\s]*rel="${key}"`, 'g');
+  const arr = regex.exec(links);
 
-const useTelescopeContributor = () => {
-  const [contributor, setContributor] = useState<Contributor | null>(null);
+  if (arr !== null) {
+    [, url, page] = arr;
+  }
+
+  return { url, page };
+};
+
+const getMaxContributors = async () => {
+  const response = await fetch(prepareURL());
```

Comment:
You can use `useSWR` for this, which will cache things nicely.  We just don't want the infinite version.

Reply:
If I `useSWR` here I might have to move this function into the hook.

Reply:
Or can I make this function a hook?

Reply:
You can do conditional fetches, or chain them with useSWR

Reply:
To be honest, this is the only place we call this hook, so I'd consider moving all of it into the component that uses it instead.